scheduler/src/cook/mesos/util.clj

```diff
@@ -489,7 +489,9 @@
 (defn create-task-ent
   "Takes a pending job entity and returns a synthetic running task entity for that job"
   [pending-job-ent & {:keys [hostname slave-id] :or {hostname nil slave-id nil}}]
-  (merge {:job/_instance pending-job-ent
+  (merge {; Synthetic ID's so that we don't incur uncachable task-ent's in task->feature-vector.
+          :instance/synthetic-id (:db/id pending-job-ent)
```

Comment:
Instead of introducing this field can the `extract-key` not look into `:job/_instance` and retrieve the id?

Reply:
Good suggestion. Done. Simplifes the code enough I rebased it.